test/spec/modules/rubiconBidAdapter_spec.js

```diff
@@ -1672,8 +1682,8 @@ describe('the rubicon adapter', function () {
           expect(
             bidderRequest.bids[0].getFloor.calledWith({
               currency: 'USD',
-              mediaType: 'video',
-              size: [640, 480]
+              mediaType: '*',
+              size: '*'
```

Comment:
why did this change?

I think we may need to overwrite the IMP processor `setOrtbImpBidFloor` if possible.

Because we split out banner and video, we know that we can call getFloor with `video` and the right size.

Not sure if we can overwrite that processor, or just call `getFloor` ourselves in the setBidFloors function?

Reply:
This decision came after a long discussion with bret that I hope to be able to summarize here... 

My question: 

> looks like the only issue still open in the adapter is about priceFloors. I don't want to reinvent the wheel since the conversion library is already generating them, but there's a difference in how we call the bidReuqest.getFloor() compared to demetrio's.
Demetrio's library is calling getFloor with {mediaType: '*', size '*'}
while we were caling it using the {mediaType: 'video', size: parseSizes(bidRequest, mediaType) } .
So the question is, are we ok with this approach, or our customers may implement different floors per every media type?
[5:57](https://magnite.slack.com/archives/D01CSGJEULW/p1665590251247539)
it looks like a single imp may have more than one mediaType, and it must contain only one floor, but we're allowing each mediaType to have it's own floor and this generates my confusion.

Response from @bretg : 

> Good catch. Tough one. Let's start with what Demetrio has. We might change this.


Michele Nasti
  [6:15 PM](https://magnite.slack.com/archives/D01CSGJEULW/p1665677737885129)
> ok, so I removed rubicon's logic and now it's on the library to get the floors, using mediatype '*' and size: '*'

Reply:
This decision came after a long discussion with bret that I hope to be able to summarize here... 

My question: 

> looks like the only issue still open in the adapter is about priceFloors. I don't want to reinvent the wheel since the conversion library is already generating them, but there's a difference in how we call the bidReuqest.getFloor() compared to demetrio's.
Demetrio's library is calling getFloor with {mediaType: '*', size '*'}
while we were caling it using the {mediaType: 'video', size: parseSizes(bidRequest, mediaType) } .
So the question is, are we ok with this approach, or our customers may implement different floors per every media type?
it looks like a single imp may have more than one mediaType, and it must contain only one floor, but we're allowing each mediaType to have it's own floor and this generates my confusion.

Response from @bretg : 

> Good catch. Tough one. Let's start with what Demetrio has. We might change this.


Michele Nasti
> ok, so I removed rubicon's logic and now it's on the library to get the floors, using mediatype '*' and size: '*'

Reply:
In that chat I asked more questions about the floors thing. I again pointed out that demetrio's getFloors was generic and not tailored for mediatype. Bret response: 

> _Actually, the getFloor function is correct. Prebid has expanded floors to be much more flexible than ORTB allows. The whole point of getFloor is to parse the crazy-complicated syntax of the Prebid floors file and return a simple number for the bidder's endpoint._

Soooooo... i stopped thinking too much about it :)

Reply:
👍

Reply:
We should prob talk to data science about this and maybe change it back in the future, but eh whatever.

We can do that later (which means we never will)